### PR TITLE
Regchest

### PR DESCRIPTION
**Pull request by Oscar-Pepper**

### CODE REVIEW

**Comment by juanky201271** [comment]
It's failing in the integration test, the problem is here: https://github.com/Oscar-Pepper/zingo-mobile/blob/97724e898da193fc6dbc1109fcdcd23bd0057749/android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt#L182
Now the `sync` command sometimes return: `task 99999 panicked`, and the JSON mapper doesn't understand this. Keep in mind that `all` commands can in some point to return a string like this: `Error: xxxxxxxxxxxxxxxxxxxxxxxxxx`.
But in this case, the `sync` process can fail with the new zingolib feature `shardtree`, 
See this: https://github.com/zingolabs/zingolib/issues/450#issuecomment-1693883516

**Comment by Oscar-Pepper** [comment]
> It's failing in the integration test, the problem is here: https://github.com/Oscar-Pepper/zingo-mobile/blob/97724e898da193fc6dbc1109fcdcd23bd0057749/android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt#L182 Now the `sync` command sometimes return: `task 99999 panicked`, and the JSON mapper doesn't understand this. Keep in mind that `all` commands can in some point to return a string like this: `Error: xxxxxxxxxxxxxxxxxxxxxxxxxx`. But in this case, the `sync` process can fail with the new zingolib feature `shardtree`, See this: [zingolabs/zingolib#450 (comment)](https://github.com/zingolabs/zingolib/issues/450#issuecomment-1693883516)

i dont have this issue locally though, if shardtree is the issue i would also fail locally right?

**Comment by juanky201271** [comment]
> > It's failing in the integration test, the problem is here: https://github.com/Oscar-Pepper/zingo-mobile/blob/97724e898da193fc6dbc1109fcdcd23bd0057749/android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt#L182 Now the `sync` command sometimes return: `task 99999 panicked`, and the JSON mapper doesn't understand this. Keep in mind that `all` commands can in some point to return a string like this: `Error: xxxxxxxxxxxxxxxxxxxxxxxxxx`. But in this case, the `sync` process can fail with the new zingolib feature `shardtree`, See this: [zingolabs/zingolib#450 (comment)](https://github.com/zingolabs/zingolib/issues/450#issuecomment-1693883516)
> 
> i dont have this issue locally though, if shardtree is the issue i would also fail locally right?

Well, depends that what commit of zingolib are you using for build the rust libs... in CI is the last dev, but in local... I think you need to check this out in your local.

File: rust/android/Cargo.toml [hunk 1]
```diff
@@ -15,7 +18,8 @@ tokio =  { version = "1.24.2", features = ["full"] }
 zingo-testutils = { git="https://github.com/zingolabs/zingolib", default-features=true, branch = "dev" }
 json = "0.12.4"
 env_logger = "0.10.0"
-test_utils = { path = "../test_utils" }
+zingomobile_utils = { path = "../zingomobile_utils" }
+regchest_utils = { git="https://github.com/Oscar-Pepper/zingo-regchest", default-features=true, branch = "launch_regtest" }
```
Comment (juanky201271) [inline rust/android/Cargo.toml]: I need to understand all of this... Why you need to refer to this forked repo?

Comment (Oscar-Pepper) [reply rust/android/Cargo.toml]: my plan was to work on my own repo and send a PR to the zingolabs repo for review... i actually messed that up by accidentally merging to zingolabs haha. its pointing at zingolabs now i have *hopefully* finished

**Review by juanky201271 — APPROVED** [review]
Really nice.

**Pull request merged**